src/core/main.c

```diff
@@ -744,6 +747,10 @@ static void set_manager_defaults(Manager *m) {
         m->default_memory_accounting = arg_default_memory_accounting;
         m->default_tasks_accounting = arg_default_tasks_accounting;
         m->default_tasks_max = arg_default_tasks_max;
+        m->default_log_max_level = arg_default_log_max_level;
+
+        if (!m->log_max_level_overridden)
+                log_set_max_level(m->default_log_max_level);
```

Comment:
actually, this is redundant, no? i mean, we do essentially the same in parse_config() already earlier, no?

Reply:
In my understanding set_manager_defaults() set the manager object attribute. At the time parse_config_file(), the manager object doesn't exist yet.

Reply:
yes, that's true. but the code above does two things: it propagates the log level from arg_default_log_level_max into the manager object, and it also updates log.c's log level. I am not questioning the former, but the latter really. I mean, log.c got updated with the log level right-away already, hence no need to update it here again

Reply:
I think it's done for the reload path: if the default value has been changed (via system.conf) and sysadmin issues "systemctl daemon-reload", we want the new value to take effect.

Reply:
ah, so i think the log level and target should probably not be set in set_manager_defaults() at all... Note that there's currently a split between set_manager_defaults() and set_manager_settings(). The former only sets default values, in the sense that the manager is not affected by them itself, it's just default values that propagate into the units loaded later. The latter otoh affects the manager itself. I am pretty sure that log level/target are not of the former kind, and hence should be added to the latter instead.

Moreover I think putting these two settings into effect should probably be in a separate function altogether, that is called at the two appropriate places.

Reply:
I see but it seems there's an issue: when reloading set_manager_settings() is not called although some manager settings might have been changed through system.conf. I submitted a new version that fixes this.